### PR TITLE
[Quantum] Output error information in table when Quantum job fails

### PR DESCRIPTION
**Pull request by warren-jones**
Modified the Quantum CLI extension code to output error information when a job fails and the CLI output format parameter "-o table" has been specified.  The error message was nested too deeply in the Quantum provider's response for normal table formatting to display it.

The new output looks like this:
```
Status    Error Code             Error Message              Target          Job ID                                Submission Time
--------  ---------------------  -------------------------  --------------  ------------------------------------  --------------------------------
Failed    InsufficientResources  Too many qubits requested  ionq.simulator  19f19add-4617-4cd6-8ec6-66d0b055d332  2022-02-25T18:56:53.275035+00:00
```
---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

### CODE REVIEW

**Comment by yonzhan** [comment]
Quantum

**Comment by warren-jones** [comment]
@vxfield, Please review

File: src/quantum/azext_quantum/commands.py [hunk 1]
```diff
@@ -94,6 +98,30 @@ def one(key, value):
...
 
+    elif 'errorData' in results:
+        notFound = 'Not found'
+        status = notFound
+        errorCode = notFound
+        errorMessage = notFound
+        target = notFound
+        jobId = notFound
+        submissionTime = notFound
+        if 'status' in results:
```
Comment (vxfield) [inline src/quantum/azext_quantum/commands.py:109]: Might be better to use [inline ternary operator](https://pythonexamples.org/python-ternary-operator/) instead of these `if` statements

**Review by vxfield — APPROVED** [review]
Approved with a suggestion

File: src/quantum/azext_quantum/commands.py [hunk 2]
```diff
@@ -94,6 +98,30 @@ def one(key, value):
...
+        if 'code' in errorData:
+            errorCode = errorData['code']
+        if 'message' in errorData:
+            errorMessage = errorData['message']
+        if 'target' in results:
+            target = results['target']
+        if 'id' in results:
+            jobId = results['id']
+        if 'creationTime' in results:
+            submissionTime = results['creationTime']
```
Comment (wangzelin007) [inline src/quantum/azext_quantum/commands.py:121]:
```suggestion
    elif 'errorData' in results:
        notFound = 'Not found'
        errorData = results['errorData']
        status = results.get('status', notFound)
        errorCode = errorData.get('code', notFound)
        errorMessage = errorData.get('message', notFound)
        target = results.get('target', notFound)
        jobId = results.get('id', notFound)
        submissionTime = results.get('creationTime', notFound)
```

File: src/quantum/azext_quantum/commands.py (hunk 1 above)
Comment (warren-jones) [reply src/quantum/azext_quantum/commands.py]: Thanks, XField, I really appreciate you pointing out the Python ternary operator, and I will definitely keep that construct in mind for future Python adventures, but in this instance I’d like to go with Zelin’s suggestion below to use the dict.get() function.

File: src/quantum/azext_quantum/commands.py (hunk 2 above)
Comment (warren-jones) [reply src/quantum/azext_quantum/commands.py]: Good call Zelin -- My original 18 lines are condensed into 6, and it still protects against raising a KeyError like my more verbose series of “if…in” statements would.  Thanks! 👍

**Pull request merged**